gobblin-metrics/src/main/java/gobblin/metrics/MetricContext.java

```diff
@@ -138,10 +138,11 @@ public String getName() {
    *
    * @param childContext the child {@link MetricContext} to add
    */
-  public void addChildContext(String childContextName, MetricContext childContext) {
-    if (this.children.putIfAbsent(childContextName, childContext) != null) {
```

Comment:
You can do `this.children.asMap().putIfAbsent()`, then you don't need to use `synchronized`.